src/index.js

```diff
@@ -30,6 +30,11 @@ export default class OfflinePlugin {
       AppCache: false
     });
 
+    if (options.ServiceWorker && options.ServiceWorker.plugins) {
```

Comment:
This should be done in `ServiceWorker.js` file

Reply:
It cannot be because it refers to `options`, not `this.options`.

Reply:
I have to clarify a bit more. It uses `deepExtend` to set up `this.options`, and `addTool` passes it to `ServiceWorker.js`. However, `deepExtend` breaks `this.options.ServiceWorker.plugins` and `ServiceWorker.js` can do nothing with it when received. Therefore it has to be fixed beforehand.

Reply:
Yeah, I see what you mean now. It should be handled in some other way though. I'll merge this and do the change to this part myself.

(tests seem to fail now because master has been updated)